src/pybind/mgr/dashboard/tools.py

```diff
@@ -122,6 +122,23 @@ def _request_log(self, logger_fn):
                       "{0:.3f}s".format(lat), length, getattr(req, 'unique_id', '-'), req.path_info)
 
 
+class MonConnectionChecker(cherrypy.Tool):
+    def __init__(self):
+        cherrypy.Tool.__init__(self, 'before_handler', self.check_connection,
+                               priority=10)
+        self.log = logging.getLogger('mon_connection_checker')
+
+    def check_connection(self):
+        self.log.error(f'Mon connection with mgr is {mgr.have_mon_connection()}')
+        if not mgr.have_mon_connection():
```

Comment:
Keep in mind that you're adding a cross-API call on a callback that's going to be invoked for every user request. In general, it's better to avoid doing much stuff on request/response handlers and keep them lightweight, since you can never (or we're not doing it yet) control the load coming from the user end.

I'd rather add some new notify event (based on a timer thread expiring/timing out after X secs not receiving any heartbeat), handle that event in the dashboard notify hook and set a Dashboard `mgr` variable to true/false.

But let's wait for the RADOS team to come up with a proposal here.